GSuite nudge purchase/skip flow to respect destination

niranjan-uma-shankar opened the pull request:
#### Changes proposed in this Pull Request

* Wrap the GSuite Nudge component in `<CheckoutContainer>`, so that clicking the Skip button on the nudge will use `handleCheckoutCompleteRedirect()` to take the user to signup destination. Presently, the Skip button action is hardcoded to a checklist destination for all scenarios.

Fixes #
https://github.com/Automattic/wp-calypso/issues/35406

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**Scenario 1**
* Go through the signup flow and add a plan(other than eCommerce) and domain to cart.
* Verify that you see the GSuite nudge and are able to complete the purchase.
* You should then be taken to flow destination.

**Scenario 2**
* On an existing account, add a new domain and GSuite subscription, and complete checkout. Verify the purchase completes. You should see the following thank you page:

<img width="804" alt="Screenshot 2019-09-03 at 11 08 36 AM" src="https://user-images.githubusercontent.com/1269602/64147874-65576a80-ce3f-11e9-8693-3046274362ee.png">


**Scenario 3**
* Verify that https://github.com/Automattic/wp-calypso/issues/35406 is fixed.

ramonjd commented:
I'm not sure I'm seeing exactly what's expected according to the test description.

**Skipping the G-Suite purchase:**

![Aug-19-2019 15-01-09](https://user-images.githubusercontent.com/6458278/63240977-5300fc80-c295-11e9-8866-1ca8686f99d2.gif)

**Purchasing E-commerce plan + `.com` domain + G-Suite**, I arrive at this page:

<img width="1000" alt="Screen Shot 2019-08-19 at 3 19 45 pm" src="https://user-images.githubusercontent.com/6458278/63241015-74fa7f00-c295-11e9-97ef-07494ac13d8d.png">

`/checkout/thank-you/kyjhtngdvacsdwfegdhfjtygkuhlukgjyfthdgrsefadwqswfegrdhtjy.wpcomstaging.com/39812062?d=gsuite`

Should I be redirected to somewhere here? I'm only shown this page, and there's nowhere for me to click to get to my dashboard.

niranjan-uma-shankar commented:
@ramonjd Thanks for the spot, updated code and ready for another review :)

> Skipping the G-Suite purchase:

Skipping should just take to flow destination. If an eCommerce plan was purchased, then skippng GSuite should take to Thank You page for the ecommerce plan.

> Purchasing E-commerce plan + .com domain + G-Suite, I arrive at this page:

This has been fixed.

I've also updated the testing instructions.

ramonjd commented:
Just testing this again. For scenario B I don't see the Thank You page at all.

After purchasing the G-Suite I'm taken to 

https://hash-7951a080691a63d4c6b100547c47b9e2e00c1a4b.calypso.live/checklist?site=cxcvxcvxcvewrwer23423wqerwe.wordpress.com


<img width="1022" alt="Screen Shot 2019-09-09 at 3 19 41 pm" src="https://user-images.githubusercontent.com/6458278/64504945-5f68fa00-d315-11e9-9329-8a7c6527ee06.png">

Shouldn't the site at least be selected as well?

Sorry, I don't know when I'll be able to test this again due to GM and parent leave!

Putting back into the ready to test column.

adamziel commented:
It looks pretty stale - I'm removing it from Zelda board from now. @niranjan-uma-shankar if you'll need more reviews feel free to drop this one into "needs review" here: https://github.com/orgs/Automattic/projects/92

p-jackson commented:
The failing desktop test is also failing on `master`

p-jackson commented:
I was going to try deploying this to fix #35406 but I'm not confident enough that I know how this is working or that I've tested it correctly so will leave.

adamziel commented:
@niranjan-uma-shankar I can see a bunch of new commits here - is this one ready to review?

niranjan-uma-shankar commented:
@adamziel Running a few tests, will add to the review board today.

niranjan-uma-shankar commented:
I have updated the code and all the tests pass.

@Automattic/zelda this is ready for review again.

niranjan-uma-shankar commented:
I tried with calypso.live link and I'm having trouble reproducing the issue. Here's what I tried:

1. Scenario 1 - going through a new signup and purchasing a Business plan shows the checklist for me:

![biz-signup](https://user-images.githubusercontent.com/1269602/70207099-b123f380-174f-11ea-833c-e47e95fc788d.gif)

2. Scenario 3 - Signup flow and purchase of eCommerce plan + .com domain + GSuite shows the Thank You page:

https://cloudup.com/c-B6WH6Nuh0

Are there any starting conditions that I'm missing to test?

andrewserong commented:
I just tried testing this out, and scenario 1 is working well for me. Scenario 3, purchasing e-commerce plan using credits and adding a domain and G Suite during checkout gave me this error... not sure if it's related? It was directly after selecting adding on G Suite:

![image](https://user-images.githubusercontent.com/14988353/70213350-38438c80-178d-11ea-8c8d-a6cac219b987.png)

p-jackson commented:
> Scenario 3, purchasing e-commerce plan using credits and adding a domain and G Suite during checkout gave me this error

Store can't be sandboxed when purchasing gsuite. My guess is because the domain is fake so can't be connected to Google.

andrewserong commented:
> Store can't be sandboxed when purchasing gsuite. My guess is because the domain is fake so can't be connected to Google.

Ah, of course! Silly me 🙇‍♂️– thanks @p-jackson!

p-jackson commented:
Figured it out! I was testing in FF, everything works in Chrome.

I'm pretty sure we ran into this issue last time we worked on the checkout @niranjan-uma-shankar, something to do with FF being stricter with the cookie we use for persisting the flow destination. The domain being tested on is `*.live` not `*.wordpress.com`.

This will work once in production.

niranjan-uma-shankar commented:
> I'm pretty sure we ran into this issue last time we worked on the checkout @niranjan-uma-shankar, something to do with FF being stricter with the cookie we use for persisting the flow destination. The domain being tested on is *.live not *.wordpress.com.

Ah yes, it's the same issue indeed :)
